grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -495,33 +515,20 @@ private void close(Status status, Metadata metadata) {
         final StatusAndMetadata statusAndMetadata = new StatusAndMetadata(status, metadata);
         logBuilder.responseContent(GrpcLogUtil.rpcResponse(statusAndMetadata, firstResponse), null);
         if (status.isOk()) {
+            req.close();
             req.abort();
         } else {
+            req.close(statusAndMetadata.asRuntimeException());
```

Comment:
This is a slight behavior change. It mimics the logic from onNext such that we close and then abort. I think it's the right thing to do in general, for when handing a Throwable thrown from other ClientCall.Listener callbacks.

Reply:
Is there material difference between close and abort? I think abort(ex) is more fitting for the exceptional case. WDYT?

Reply:
I poked around and settled on calling abort in case of unhappy closure - I think it's the right thing to do. I also added [a comment](https://github.com/line/armeria/pull/4851/files#diff-39112577712e2be6800f84834abc33a690eed333cd96581b777e1e1cd1ccdadaR544-R549) explaining some reasoning behind it. PTAL.